panther_gpiod/include/panther_gpiod/gpio_driver.hpp

```diff
@@ -0,0 +1,265 @@
+   * This method sets the provided callback function to be executed upon GPIO edge events.
+   *
+   * @param callback The callback function to handle GPIO edge events.
+   *
+   * @par Example
+   * An example of using this method to bind a member function as a callback:
+   * @code{.cpp}
+   * class MyClass {
+   * public:
+   *   void handle_gpio_event(const GPIOinfo & gpio_info) {
+   *     // Handle GPIO event here, i.e:
+   *     std::cout << gpio_info.offset << ":    " << gpio_info.value << std::endl;
+   *   }
+   * };
+   *
+   * MyClass my_obj;
+   * GPIODriver gpio_driver;
+   * gpio_driver.configure_edge_event_callback(
+   *     std::bind(&MyClass::handle_gpio_event, &my_obj, std::placeholders::_1));
+   * @endcode
+   */
+  void configure_edge_event_callback(const std::function<void(const GPIOinfo &)> & callback)
+  {
+    gpio_edge_event_callback = callback;
+  }
+
+  /**
+   * @brief Checks if a specific GPIO pin is active.
+   *
+   * This method returns the value stored in the class read during the last edge event.
+   *
+   * @param pin GPIOpin to check.
+   * @return True if the pin is active, false otherwise.
+   */
+  bool is_pin_active(GPIOpin pin);
+
+  /**
+   * @brief Sets the value for a specific GPIO pin.
+   *
+   * @param pin GPIOpin to set the value for.
+   * @param value New value to set for the pin.
+   * @return True if the value is set successfully, false otherwise.
+   */
+  bool set_pin_value(GPIOpin pin, bool value);
+
+  /**
+   * @brief Changes the direction of a specific GPIO pin.
+   *
+   * @param pin GPIOpin to change the direction for.
+   * @param direction New direction for the pin.
+   * @throws std::runtime_error if there is an error while setting the GPIO pin value.
+   */
+  void change_pin_direction(GPIOpin pin, gpiod::line::direction direction);
+
+  /**
+   * @brief Enables asynchronous monitoring of GPIO pin events.
+   */
+  void gpio_monitor_on();
+
+  /**
+   * @brief Disables asynchronous monitoring of GPIO pin events.
+   */
+  void gpio_monitor_off();
+
+private:
+  std::unique_ptr<gpiod::line_request> create_line_request(gpiod::chip & chip, const GPIOpin pin);
+  std::unique_ptr<gpiod::line_request> create_line_request(
+    gpiod::chip & chip, const std::vector<GPIOpin> & pins);
+  gpiod::line_settings generate_line_settings(const GPIOinfo & pin_info);
+  GPIOpin get_pin_from_offset(gpiod::line::offset offset) const;
+  GPIOinfo & get_pin_info_ref(GPIOpin pin);
+  void configure_line_request(gpiod::chip & chip, gpiod::request_builder & builder, GPIOpin pin);
+  void monitor_async_events();
+  void handle_edge_event(const gpiod::edge_event & event);
+  bool is_gpio_monitor_thread_running() const;
+
+  /**
+   * @brief Callback function for GPIO edge events.
+   *
+   * @param gpio_info Information related to the state of the GPIO pin for which the event took
+   * place.
+   */
+  std::function<void(const GPIOinfo & gpio_info)> gpio_edge_event_callback;
```

Comment:
IMO it can be justified by the need to provide internal documentation, even if these fields are not visible to clients or inheriting classes, aiding in code comprehension, maintenance, and adherence to coding conventions.